src/main/webapp/script.js

```diff
@@ -456,13 +475,17 @@ function buildSuggestedLocationWidget(name, vicinity, photoSrc) {
   addressText.className = 'card-title';
   addressText.innerText = vicinity;
 
+  // Create the card footer, where action items ("Add this POI") will be placed.
+  const cardFooter = document.createElement('div');
+  cardFooter.className = 'card-footer';
+
   // Add a button that allows you to add this suggested location as a POI.
   const addSuggestedPoiButton = document.createElement('button');
   addSuggestedPoiButton.className = 'btn btn-primary';
   addSuggestedPoiButton.innerText = 'Add this POI';
   addSuggestedPoiButton.onclick = () => {
     // Add this POI.
-    addPoi(name);
+    addPoi(name + ', ' + vicinity);
```

Comment:
This change was made to clarify the POI. For locations like "Lincoln Park", the backend couldn't tell which "Lincoln Park" the user was referring to. Adding the vicinity solves this problem.